### PR TITLE
Cleanup patches are as consistent as possible with the bambu studio master branch

### PR DESCRIPTION
**Pull request by MackBambu**
Bambu master branch receives partial patches

### CODE REVIEW

**Comment by MackBambu** [comment]
I created two repositories to receive and test if the patches work.

File: com.bambulab.BambuStudio.yml [hunk 1]
```diff
@@ -98,19 +98,19 @@ modules:
...
-      - ./b2 headers
-      - ./b2 -j$FLATPAK_BUILDER_N_JOBS install variant=release cxxstd=17 --layout=system
-    sources:
-      - type: archive
-        url: https://freefr.dl.sourceforge.net/project/boost/boost/1.78.0/boost_1_78_0.zip
-        sha256: f22143b5528e081123c3c5ed437e92f648fe69748e95fa6e2bd41484e2986cc3
-    cleanup:
-      - /include
-      - /lib/cmake
+  # - name: boost
```
Comment (hadess) [inline com.bambulab.BambuStudio.yml:101]: Why would we want to build boost inside the BambuStudio module rather than separately, it just makes the BambuStudio build longer.

File: com.bambulab.BambuStudio.yml [hunk 2]
```diff
@@ -144,7 +144,7 @@ modules:
       - -DwxUSE_OPENGL=ON
       - -DwxUSE_WEBREQUEST=ON
       - -DwxUSE_WEBVIEW=ON
-      - -DwxUSE_REGEX=sys
+      - -DwxUSE_REGEX=builtin
```
Comment (hadess) [inline com.bambulab.BambuStudio.yml:147]: Why? This should be a separate commit with a description of why the change was made.

File: com.bambulab.BambuStudio.yml [hunk 3]
```diff
@@ -156,18 +156,9 @@ modules:
...
-      - type: patch
-        path: patches/disable-gstplayer.patch
-      # https://github.com/wxWidgets/wxWidgets/issues/23630
-      - type: patch
-        path: patches/0001-Add-support-for-building-WebView-with-libwebkit2gtk-.patch
-      # wx-config/build fixes between 3.1.5 and 3.1.6
-      - type: patch
-        path: patches/wxwidgets-not-found.patch
+      - type: git
+        url: https://github.com/MackBambu/wxWidgets
```
Comment (hadess) [inline com.bambulab.BambuStudio.yml:160]: We can't use this repo, this isn't an official wxWidgets repo.

File: com.bambulab.BambuStudio.yml [hunk 4]
```diff
@@ -228,6 +219,16 @@ modules:
         dest: external-packages/Blosc
         sha256: dcb48bf43a672fa3de6a4b1de2c4c238709dad5893d1e097b8374ad84b1fc3b3
 
+      - type: file
```
Comment (hadess) [inline com.bambulab.BambuStudio.yml:222]: Building more stuff in the BambuStudio module makes the whole build take longer if we need to rebuild, because all the deps will be rebuilt as well.

As it is, if the boost or any dependencies before that aren't changed, we'll be rebuilding BambuStudio and all of its internal deps. Building more stuff inside the BambuStudio module as part of its internal deps is not desirable.

Patches are also more difficult to apply to the BambuStudio internal deps than if we were building them separately.

File: com.bambulab.BambuStudio.yml [hunk 5]
```diff
@@ -314,20 +315,20 @@ modules:
       - type: patch
         path: patches/0001-Apply-TBB-LTO-patch.patch
 
-      - type: patch
-        path: patches/0001-deps-Cereal-Update-Cereal-to-fix-build-error.patch
+      # - type: patch
+      #   path: patches/0001-deps-Cereal-Update-Cereal-to-fix-build-error.patch
```
Comment (hadess) [inline com.bambulab.BambuStudio.yml:319]: Patches removed with no justification.

**Review by hadess — CHANGES_REQUESTED** [review]
This can't be merged in this state.

**Comment by hadess** [comment]
> I created two repositories to receive and test if the patches work.

I don't understand what this means. If you're just doing build testing, it would be great if you could run those tests locally before sending merge requests that are unmergeable.

File: com.bambulab.BambuStudio.yml [hunk 6]
```diff
@@ -302,8 +303,8 @@ modules:
 
       # BambuStudio Source Archive
       - type: git
-        url: https://github.com/bambulab/BambuStudio.git
-        tag: v01.09.01.66
+        url: https://github.com/MackBambu/BambuStudio.git
```
[the remark this answers is not in the document]
Comment (MackBambu) [reply com.bambulab.BambuStudio.yml:306]: This is just a test branch, if everything is stable I'll merge the changes into the master branch.

**Comment by MackBambu** [comment]
> I don't understand what this means. If you're just doing build testing, it would be great if you could run those tests locally before sending merge requests that are unmergeable.

Don't worry, I'm in no hurry to merge code.
My current job is to merge some patches into the bambu master branch, but it needs a lot of testing before that.
This PR still needs a lot of changes with your help.

File: com.bambulab.BambuStudio.yml (hunk 5 above)
Comment (MackBambu) [reply com.bambulab.BambuStudio.yml:319]: https://github.com/MackBambu/BambuStudio/commit/792df7265bf489e9a0d0aa5c44e7266b1b05322e

**Comment by MackBambu** [comment]
@hadess Is there any flag to detect the flatpak compilation environment in cmake? I'm trying to merge use-sysdeps.patch into Bambu studio so we can drop this.

**Comment by hadess** [comment]
> @hadess Is there any flag to detect the flatpak compilation environment in cmake? I'm trying to merge use-sysdeps.patch into Bambu studio so we can drop this.

You can set any environment variable in the Flatpak manifest, look for `env` in the `flatpak-manifest` man page.

I think a better way to do this would be to add specific CMake options to allow `sys` dependencies for all the internal dependencies we disable in that patch, like wxwidgets allows:
```
  - name: wxwidgets
    buildsystem: cmake-ninja
    config-opts:
      - -DwxBUILD_PRECOMP=ON
      - -DwxBUILD_TOOLKIT=gtk3
      - -DwxBUILD_DEBUG_LEVEL=0
      - -DwxBUILD_SAMPLES=OFF
      - -DwxBUILD_SHARED=OFF
      - -DwxUSE_MEDIACTRL=ON
      - -DwxUSE_DETECT_SM=OFF
      - -DwxUSE_UNICODE=ON
      - -DwxUSE_PRIVATE_FONTS=1
      - -DwxUSE_OPENGL=ON
      - -DwxUSE_WEBREQUEST=ON
      - -DwxUSE_WEBVIEW=ON
      - -DwxUSE_REGEX=sys
      - -DwxUSE_LIBSDL=OFF
      - -DwxUSE_XTEST=OFF
      - -DwxUSE_STC=OFF
      - -DwxUSE_AUI=ON
      - -DwxUSE_LIBPNG=sys
      - -DwxUSE_ZLIB=sys
      - -DwxUSE_LIBJPEG=sys
      - -DwxUSE_LIBTIFF=sys
      - -DwxUSE_NANOSVG=OFF
      - -DwxUSE_EXPAT=sys
```

If we had this for libpng, libjpeg, tiff, boost, etc. we could drop the patch and set some config options instead.

**Comment by MackBambu** [comment]
> If we had this for libpng, libjpeg, tiff, boost, etc. we could drop the patch and set some config options instead.

Look at this.
https://github.com/MackBambu/BambuStudio/commit/3c4a307a64fb621cf830d5941ec2fd006eb7d088

**Comment by hadess** [comment]
> Look at this.
> [MackBambu/BambuStudio@3c4a307](https://github.com/MackBambu/BambuStudio/commit/3c4a307a64fb621cf830d5941ec2fd006eb7d088)

Much better. Let me know when you've submitted this in an MR, and I'll review it.

**Comment by hadess** [comment]
It would be great if you could try not to use this shared resource as a build server. Please build your changes locally (you can even reference local `file:///` paths for git, so that you don't have to push your changes to the server).

The "close GLFW build TEST and EXAMPLES" commit you pushed is already in this repository under the subject "Disable docs building in glfw", you'll need to rebase your branch to get it.

**Comment by MackBambu** [comment]
> It would be great if you could try not to use this shared resource as a build server. Please build your changes locally (you can even reference local `file:///` paths for git, so that you don't have to push your changes to the server).
> 
> The "close GLFW build TEST and EXAMPLES" commit you pushed is already in this repository under the subject "Disable docs building in glfw", you'll need to rebase your branch to get it.

I actually built everything fine locally by running local-build.sh. :(

图片

**Comment by MackBambu** [comment]
Why are the results different between server and local builds?

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master/
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer
Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master
Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

Here are the wxWidgets after merging the patches:
https://github.com/bambulab/wxWidgets/commits/master

Here's the studio commit for the merged patch:
https://github.com/bambulab/BambuStudio/commit/c8378296fa42836408dceb877452b7d4aa32c1f6
https://github.com/bambulab/BambuStudio/commit/a181250668fae3110e329c0dd289f1e71dd1bb45

**Comment by MackBambu** [comment]
@hadess I think it's ready. please reviewer

**Comment by hadess** [comment]
This isn't in a reviewable state. Please rework your changes to include one justified change per commit message.

I'm not that interested in relying on vendor forks of the upstream libraries, as I'd rather we shipped a small number of patches that we could reduce over time.

The right thing to do, in wxWidgets, is to make BambuStudio use the latest versions of wXwidgets, and fix the bugs that were found upstream.

**Comment by hadess** [comment]
Also note that we only ship releases of BambuStudio, so we wouldn't be able to merge this as it would require shipping the current development version.

**Comment by MackBambu** [comment]
> The right thing to do, in wxWidgets, is to make BambuStudio use the latest versions of wXwidgets, and fix the bugs that were found upstream.

wxWidgets won't be upgraded anytime soon, this could be a slow process.
You can check out similar projects like Prusa Studio, Orca Slicer, which is common practice.

**Comment by hadess** [comment]
> wxWidgets won't be upgraded anytime soon, this could be a slow process.
> You can check out similar projects like Prusa Studio, Orca Slicer, which is common practice.

I'm not interested in relying on a vendored dependency in the Flatpak, I'd rather ship the upstream version with patches, it's really not that big a deal for our packaging.

**Comment by MackBambu** [comment]
The master branch of Bambu Studio has switched to relying on the forked version of wxWidget, so I don't see what's wrong with that?
Shouldn't the dependency sources be consistent?

**Comment by MackBambu** [comment]
> Also note that we only ship releases of BambuStudio, so we wouldn't be able to merge this as it would require shipping the current development version.

1. Release 1.9.2
2. Merge my PR
3. Wait release 1.9.3

**Comment by hadess** [comment]
Please rework the commits as requested before we can discuss individual changes.

**Comment by MackBambu** [comment]
> Please rework the commits as requested before we can discuss individual changes.

ok, I'll reorganise the commits when we release 1.9.3.
As for the wxWidgets source, I think you have to think about it a bit more, we'll keep changing wxWidgets in the future and your patch file will get bigger and bigger.

**Comment by hadess** [comment]
Closing in favour of https://github.com/flathub/com.bambulab.BambuStudio/pull/28
